spyder/widgets/ipythonconsole/shell.py

```diff
@@ -298,6 +318,12 @@ def handle_exec_method(self, msg):
                     else:
                         properties = None
                     self.sig_var_properties.emit(properties)
+                elif 'get_cwd' in method:
+                    if data is not None and 'text/plain' in data:
+                        self._cwd = ast.literal_eval(data['text/plain'])
+                    else:
+                        self._cwd = None
```

Comment:
This needs to be

    self._cwd = ''

because `sig_change_cwd` expects a string (because of its signature). This is what's causing the segfaults seen in our tests.

Strangely enough, PyQt4 and PyQt 5.7+ (i.e. pip wheels) coerces `None` to `str`, but that's not the case for PyQt 5.6 (used by Anaconda).

With this change all tests pass locally for me (God bless our previous tests!! This would have been be a very serious bug in 3.2 ;-)